### PR TITLE
fast* utils

### PR DESCRIPTION
**Pull request by Thomazella**
Faster and browser compatible alternatives to Array.prototype functions. (Really WIP guys)

### CODE REVIEW

File: packages/reakit/src/_utils/fastFilter.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const fastFilter = (func, list) => {
```
Comment (diegohaz) [inline packages/reakit/src/_utils/fastFilter.js]: I'd use `fastFilter = (list, func)`

File: packages/reakit/src/_utils/fastMap.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const fastMap = (func, list) => {
```
Comment (diegohaz) [inline packages/reakit/src/_utils/fastMap.js]: Same.

File: packages/reakit/src/_utils/fastReduce.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+const fastReduce = (func, initial, list) => {
```
Comment (diegohaz) [inline packages/reakit/src/_utils/fastReduce.js]: `list, initial, func`

File: packages/reakit/src/_utils/fastFilter.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  const result = new Array(tempIndex);
+
+  while (index < tempIndex) {
+    result[index] = temp[index];
+    index += 1;
+  }
+  return result;
+};
+
+module.exports = fastFilter;
```
Comment (diegohaz) [inline packages/reakit/src/_utils/fastFilter.js]: Let's use `export default`.

File: packages/reakit/src/_utils/fastReduce.js (hunk 3 above)
Comment (diegohaz) [reply packages/reakit/src/_utils/fastReduce.js]: Actually, since `initial` can be omitted, it should be `list, func, initial`

Comment (diegohaz) [reply packages/reakit/src/_utils/fastReduce.js]: So it follows the order: `list.reduce(func, initial)`

File: packages/reakit/src/_utils/fastFilter.js (hunk 4 above)
Comment (rthomazel) [reply packages/reakit/src/_utils/fastFilter.js]: It wasn't working for me, very strange

Comment (diegohaz) [reply packages/reakit/src/_utils/fastFilter.js]: If you're importing it with commonjs (`require`), you'll need to add `.default` to the required object.

**Comment by Thomazella** [comment]
It's old

**Pull request closed without merging**